### PR TITLE
Bump version to 0.8.0

### PR DESCRIPTION
**Pull request by pquentin**
I had to copy the `misc` entries manually since it's a [non-standard category](https://github.com/python-trio/trio/blob/e930d11dd294c60a8c9223bf1ba4e800a8e6cc24/pyproject.toml#L48-L51).

Sorry for pushing this branch to the main repo, my remotes were not configured correctly.

### CODE REVIEW

File: docs/source/history.rst [hunk 1]
```diff
@@ -5,6 +5,44 @@ Release history
 
 .. towncrier release notes start
 
+pytest-trio 0.8.0 (2022-11-01)
+------------------------------
+
+Features
+~~~~~~~~
+
+- If a test raises an ``ExceptionGroup`` (or nested ``ExceptionGroup``\ s) with only
```
Comment (oremanj) [inline docs/source/history.rst:14]: Does this definitely render right? I would write `\s` with no space here, and skip the `\ ` before a comma, but my sphinx-fu is not elevated enough to know whether these will actually be parsed differently.

Comment (oremanj) [inline docs/source/history.rst:14]: Could probably make ExceptionGroup a link? (intersphinx to the Python docs)

**Review by oremanj — APPROVED** [review]

File: docs/source/history.rst (hunk 1 above)
Comment (pquentin) [reply docs/source/history.rst:14]: It does render right: https://pytest-trio--131.org.readthedocs.build/en/131/history.html

Well, https://pytest-trio.readthedocs.io/en/stable/history.html is up now already.

Comment (pquentin) [reply docs/source/history.rst:14]: Sorry, our release process mentions that no review is needed, and while this is a valid improvement, I uploaded the release to PyPI already. Would be happy to accept a PR from anyone that fixes that, even though it won't warrant a new release.

**Pull request merged**